packages/datagrid/src/datagrid.ts

```diff
@@ -2096,6 +2096,54 @@ export class DataGrid extends Widget {
     MessageLoop.postMessage(this._viewport, Private.OverlayPaintRequest);
   }
 
+  private _getMaxWidthInBodyColumn(index: number): number | null {
+    const dataModel = this.dataModel;
+
+    if (!dataModel) {
+      return null;
+    }
+
+    const numRows = dataModel.rowCount('body');
+
+    let maxWidth = 0;
+    for (let j = 0; j < numRows; j++) {
+      const textWidth = this._getCellTextWidth(dataModel, index, j, 'body');
+      maxWidth = Math.max(maxWidth, textWidth);
+    }
+
+    return maxWidth;
+  }
+
+  private _getCellTextWidth(
```

Comment:
This is reusing existing code to find the length of a cell in pixels.

This doesn't seem accurate. Do you have a recommended way for getting the length of any random Renderer? I'm not very familiar with the abstractions.

Reply:
It may need setting more context options for higher fidelity. For reference eliding uses this:

https://github.com/jupyterlab/lumino/blob/129221f9959d0f92e0c69159b464bd23cbc1f0a3/packages/datagrid/src/textrenderer.ts#L210-L217

Reply:
Thanks for the suggestion! This seems to help!